pkg/geo/geogfn/topology_operations.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geogfn
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/errors"
+	"github.com/golang/geo/r3"
+	"github.com/golang/geo/s2"
+	"github.com/twpayne/go-geom"
+)
+
+// Centroid returns the Centroid of a given Geography.
+//
+// Note: In the case of (Multi)Polygon Centroid result, don't mirror with
+// `PostGIS`'s result. Though we are using the same algorithm. Their
```

Comment:
It looks like we're choosing the same starting point (the first one) here.
Why is this different? It's not 100% clear to me from this comment.
Maybe worth elaborating on the algorithm instead of just saying "the same as"

Reply:
Thank you for pointing it out. I Think The comment was not clear, I have explicitly mentioned the algorithm used for calculation of Centroid in brief and highlighted what exactly meant by different implement and why we did this way.


Though we are using the same starting point as done in `PostGIS`. The difference is that we are updating reference point when moving for one Ring to another. Hence creating a triangle with a smaller area.